keps/sig-api-machinery/0006-apply.md

```diff
@@ -194,8 +195,41 @@ atomic. That can be specified with `// +mapType=atomic` or `//
 +structType=atomic` respectively. They map to the same openapi extension:
 `"x-kubernetes-map-type": "atomic"`.
 
-#### Kubectl
-##### Server-side Apply
+#### Clients
+
+##### Apply Status Header
+
+Client side apply in kubectl determines if the applied object would result in any changes on the apiserver and provides the user with this information.
+
+To provided similar information with server-side apply, a dedicated header is set by the apiserver
+if a **successful apply request** caused changes to the existing resource.
+
+```txt
+X-Kubernetes-Apply-Success-Type: "changes-made" || "changes-unneeded"
+```
+
+The value may be,
+
+- `changes-unneeded` in cases where the request did not cause a change to the resource's managedFields.
+This means changes to fields that don't get owned or get reset, will be ignored when setting the header.
+- `changes-made` in cases where the request causes a change to the resource's managedFields.
+
+This header is optional.
+If the header is missing on the response, the apiserver may be older than this feature.
+Clients should never deduce anything from the absence of this header.
+
+This header can then be used by e.g. `kubectl` to provide the user with the desired information (rel: [#85750](https://github.com/kubernetes/kubernetes/issues/85750), [#66450](https://github.com/kubernetes/kubernetes/issues/66450)).
+
+**The header is intended to be informational only.**
+No behavior should be determined based on it's existence or value.
+
+It is not reliable to use this header as a mechanism for controlling multi-step deployments, which would be highly discouraged.
```

Comment:
```suggestion
This header is informational, not a mechanism for controlling multi-step deployments. Since clients may crash and retry from arbitrary points in a multi-step deployment, lack of action at one stage doesn't imply other stages don't need action.
```